src/Service/ActivityPubManager.php

```diff
@@ -789,4 +794,44 @@ public function getEntityObject(string|array $apObject, array $fullPayload, call
 
         return $this->entityManager->getRepository($activity['type'])->find((int) $activity['id']);
     }
+
+    public function extractMarkdownSummary(array $apObject): ?string
+    {
+        if (isset($apObject['source']) && isset($apObject['source']['mediaType']) && isset($apObject['source']['content']) && ApObjectExtractor::MARKDOWN_TYPE === $apObject['source']['mediaType']) {
+            return $apObject['source']['content'];
+        } else {
+            $converter = new HtmlConverter(['strip_tags' => true]);
+
+            return stripslashes($converter->convert($apObject['summary']));
+        }
+    }
+
+    public function extractRules(string &$description): ?string
+    {
+        $lines = explode("\n", $description);
+        $headers = [];
+        $i = 0;
+        foreach ($lines as $line) {
+            if (preg_match("/^#+ ([^\n\r]*)/", $line, $matches)) {
+                if (2 === \sizeof($matches)) {
+                    $headers[] = ['line' => $i, 'header' => $matches[1]];
+                }
+            }
+            ++$i;
+        }
+
+        $rules = null;
+
+        if (!empty($headers)) {
+            $lastHeader = $headers[\sizeof($headers) - 1];
+            if (str_contains(strtolower($lastHeader['header']), 'rules')) {
+                $descLines = \array_slice($lines, 0, $lastHeader['line']);
+                $ruleLines = \array_slice($lines, $lastHeader['line'] + 1, \sizeof($lines) - $lastHeader['line']);
+                $description = join("\n", $descLines);
+                $rules = join("\n", $ruleLines);
+            }
+        }
+
+        return $rules;
+    }
```

Comment:
I just don't think this is a good idea vs parsing it as a regular description like we've been doing, it seems way too prone to failure. For instance a random remote magazine might have a quote in its description like `## “Know the rules well, so you can break them effectively.”  - Dalai Lama XIV` we suddenly parse that as the rules. Also, it doesn't work for non-english, so is flaky

Reply:
Before thinking to hard about this what do you think about my ideas for the rules field? Because if you think it is too complicated and not really worth it then we will just deprecate the field (if everyone agrees)

Reply:
I think it's fine to keep it if you see value

My opinions on keeping it would be:
- Add it as "Rules" as you have now on outgoing
- Remove this extract rules from remote, just keep it all in description for now, even from other mbin instances
- Make an FEP for adding a rules field ordered list to group collections, if that's what you believe in